### DISCUSSION TITLE
Document upload on Web UI doesn't work behind ingress-nginx

### DISCUSSION THREAD

**Discussion by sysedwinistrator** [jonaswinkler/paperless-ng#1414] [Support]
Hi,
I have set up paperless-ng in a Kubernetes cluster, using ingress-nginx ([this one](https://github.com/kubernetes/ingress-nginx)) as reverse proxy. 
Everything seems to work, but when I try to upload files on the Web UI, the progress bar gets stuck and eventually fails with `HTTP-Fehler: 0 Unknown Error` (in red in the upload field). Developer tools in browser (Chromium and Firefox) show that the POST request is not getting any response. Chromium shows a `net::ERR_HTTP2_PROTOCOL_ERROR` error message in the console, which - according to a quick search - is a symptom of the server not answering.

But when I access the UI via `kubectl port-forward` (if you don't know kubernetes: that's the equivalent of docker publish), it works. **It also works behind a standalone HTTP NGINX reverse proxy configured as per the documentation**, which is why I am not opening a bug report. I also unsuccessfully tested the ingress without HTTPS to rule out SSL issues. It clearly is _something_ ingress-nginx does. I've been comparing the generated nginx config the ingress controller uses with the one the working standalone reverse proxy uses in the hope finding the differing option(s) that cause(s) the issue.

I've tried a lot of stuff already, including:
- Disabling HTTP2 for the ingress controller
- Enabling `proxy_buffering`
- Disabling `proxy_request_buffering`
- Changing the Connection proxy header to `"keep-alive"` or `"upgrade"`
- Increasing `client_max_body_size`
- Setting `large_client_header_buffers`
- Increasing `proxy_buffer_send_timeout` and `proxy_buffer_read_timeout` to 3600 seconds, which is recommended for WebSockets

What I might try next is to get a HTTP access log for gunicorn. The failed uploads trigger no log messages, not even with `PAPERLESS_DEBUG=true`.

**Comment by hamid-elaosta** [comment]
@sysedwinistrator Did you ever resolve this? I have the exact same issue.
